### PR TITLE
Reading events info from ceo file

### LINKED ISSUE #9380: mne.io.read_raw_curry() cannot read the event file end with '.cdt.ceo'
#### Describe the bug
I have tried to use `mne.io.read_raw_curry()` to read my data created by Curry 8, but there were no annotations in the Raw object. But EEGLAB can import the data and show the annotations correctly.

#### Steps to reproduce
I read the source code of [curry.py](https://github.com/mne-tools/mne-python/blob/maint/0.23/mne/io/curry/curry.py), and I found this function cannot handle the ".cdt.ceo" file. 

So I change the file extension of my file from ".cdt.ceo" to "cdt.cef", and the function works smoothly.

#### Expected results
`mne.io.read_raw_curry()` can read cef file too, or it will make some users confused.

#### Additional information
I am sorry for my weak program skill. I have tried to fix this bug.

### PR DESCRIPTION
**Pull request by dddd1007**
#### Reference issue
Fixes #9380

#### What does this implement/fix?
The read_raw_curry() cannot read the event file from .ceo file. So I try to make the function recognize .ceo files, a type of event file generated by Curry.

### CODE REVIEW

**Comment by drammock** [comment]
@dddd1007 I've pushed a commit here that I think will fix your problem.  Can you `git pull` this change and test it out?

**Comment by dddd1007** [comment]
@drammock, I pulled this commit and tested it. `read_raw_curry()` can read all files correctly. 

Thanks for your fixation!

**Review by agramfort — APPROVED** [review]
diff looks good

I don't know how easy it would be to test this.

it will need a what's new entry too

however it's weird to open a PR first to maint branch. I think it should be merged to main and then backported

thx @dddd1007 and @drammock

**Comment by larsoner** [comment]
Changed the base branch and closed-reopened to restart PRs. But we have a bunch of PyVista-related errors so maybe we should fix those first, then do another restart here

**Comment by larsoner** [comment]
@dddd1007 can you add an entry to `changes/latest.inc` while we wait for CIs to be fixed in #9386 and #9341?

**Comment by dddd1007** [comment]
> @dddd1007 can you add an entry to `changes/latest.inc` while we wait for CIs to be fixed in #9386 and #9341?

OK, I'll do it.

**Review by larsoner — APPROVED** [review]

File: doc/changes/latest.inc [hunk 1]
```diff
@@ -36,6 +38,7 @@ Bugs
 ~~~~
 - Fix bug with :meth:`mne.Epochs.crop` and :meth:`mne.Evoked.crop` when ``include_tmax=False``, where the last sample was always cut off, even when ``tmax > epo.times[-1]`` (:gh:`9378` **by new contributor** `Jan Sosulski`_)
 
+- Fix bug with `mne.io.read_raw_curry` when Curry 8 generate event file end with '.cdt.ceo', the function cannot read it (:gh:`9381` **by new contributor** |Xiaokai Xia|_)
```
Comment (larsoner) [inline doc/changes/latest.inc:41]:
```
changes/latest.inc:41: WARNING: Unknown target name: "xiaokai xia".
```
Can you add your name + URL to `doc/changes/names.inc`?

**Review by drammock — APPROVED** [review]
merge when CIs green.

**Comment by larsoner** [comment]
Thanks @dddd1007 !

**Pull request merged**